contracts/DxMgnPool.sol

```diff
@@ -6,22 +6,38 @@ import "./interfaces/IDutchExchange.sol";
 
 contract DxMgnPool {
     struct Participation {
-        uint startAuctionId; // dx auction ID this participation started contributing in
+        uint startAuctionCount; // how many auction passed when this participation started contributing
         uint poolShares; // number of shares this participation accounts for (absolute)
         uint amount; // Amount of depostTokens in this participation
     }
 
     mapping (address => Participation[]) public participationsByAddress;
     uint public totalPoolShares = 0;
+    uint public totalPoolSharesCummulative = 0;
     uint public totalDeposit = 0;
+    uint public totalMgn = 0;
+    uint public lastParticipatedAuctionIndex = 0;
+    uint public auctionCount = 0;
+    
     ERC20 public depositToken;
     ERC20 public secondaryToken;
+    ERC20 public mgnToken;
```

Comment:
can you use here, now or later the real MGN-Inteface?
Just import this one.
https://github.com/gnosis/dx-contracts/blob/master/contracts/TokenFRT.sol

But I am fine, if we do that later...

Reply:
Do we need the FRT specific functions? Otherwise, I'd prefer to keep the abstraction as general as possible (ERC20).

Reply:
Later on I have to unlock mgn tokens, yes